### ISSUE TITLE
`do_tests` crashes with the `--tags` option

### ISSUE THREAD

**Issue by imaliyov** [perturbo-code/perturbopy#72]
Within perturbo/tests:
```
./run_tests.py --tags dynamics
...
/global/homes/i/imaliyov/.conda/envs/perturbopy/lib/python3.10/site-packages/pluggy/_manager.py:112: in _hookexec
    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
/global/homes/i/imaliyov/.conda/envs/perturbopy/lib/python3.10/site-packages/perturbopy/conftest.py:119: in pytest_generate_tests
    test_list = filter_tests(
/global/homes/i/imaliyov/.conda/envs/perturbopy/lib/python3.10/site-packages/perturbopy/test_utils/run_test/run_utils.py:195: in filter_tests
    test_list.remove(test_name)
E   ValueError: list.remove(x): x not in list
=================================== short test summary info ====================================
ERROR ../../../../../../../homes/i/imaliyov/.conda/envs/perturbopy/lib/python3.10/site-packages/perturbopy/testing_code/test_functions.py - ValueError: list.remove(x): x not in list
!!!!!!!!!!!!!!!!!!!!!!!!!!!! Interrupted: 1 error during collection !!!!!!!!!!!!!!!!!!!!!!!!!!!!
```

**Comment by imaliyov** [comment]
@hurricane642

**Comment by hurricane642** [comment]
Solved in #73
